react/features/base/settings/reducer.js

```diff
@@ -0,0 +1,125 @@
+// @flow
+
+import { APP_WILL_MOUNT } from '../../app';
+import JitsiMeetJS, { browser } from '../lib-jitsi-meet';
+import { ReducerRegistry } from '../redux';
+import { PersistenceRegistry } from '../storage';
+
+import { SETTINGS_UPDATED } from './actionTypes';
+
+const logger = require('jitsi-meet-logger').getLogger(__filename);
+
+/**
+ * The default/initial redux state of the feature {@code base/settings}.
+ *
+ * @type Object
+ */
+const DEFAULT_STATE = {};
+
+const STORE_NAME = 'features/base/settings';
+
+/**
+ * Sets up the persistence of the feature {@code base/settings}.
+ *
+ * Note: There are some values inherited from the old Settings.js that needs
+ * session-based storage but not persisted storage, such as cameraDeviceId
+ * and micDeviceId so we utilize the selective storage functionality of the
+ * persistency engine.
+ */
```

Comment:
Do you think it's possible to make the behavior consistent right now? I'm looking at conference#_initDeviceList and honestly cannot understand how not persisting the device ids in local storage does some kind of syncing of states between features.

Reply:
As I cannot understand it...no, I don't have suggestions that should be taken now. Have you tried not supporting it at all? If you haven't maybe it's something you or I could try after this PR is merged in.

Reply:
Yeah, let's move this off to another PR, and also the settings v3 (or 4 or 5?) where everything is aggregated into a runtime place (e.g. local participant) and we can hide all these settings/jwt/config.js/urlparam mess